skyfield/sgp4lib.py

```diff
@@ -100,6 +151,10 @@ def _observe_from_bcrs(self, observer):
         # g.distance = euclidian_distance
         return g
 
+    @property
+    def semi_major_axis(self):
+        return Distance(m=(GE / (self.mean_motion ** 2)) ** (1/3))
```

Comment:
This formula brings up a big issue that at some point we should tackle. (But maybe it's too big an issue to look at right now, and we could circle back once these first pull requests are merged?) The issue is: that the semi-major axis is already computed in the SGP4 library, but it might not get saved to the `satrec` object in a way that we can use.

The code is a mess because I just barely changed the C into Python, but here is the computation of the semi-major axis that this satellite object will already have done over in the `sgp4` library:

https://github.com/brandon-rhodes/python-sgp4/blob/master/sgp4/propagation.py#L1088

https://github.com/brandon-rhodes/python-sgp4/blob/master/sgp4/propagation.py#L1152

On the one hand, it looks like several of the features and methods and attributes you want to add could be added all the way down at the `sgp4` library level, and make it a much nicer library even for people who want to use it alone without Skyfield.

On the other hand, it is such terrible code that I maybe cannot ask you to read it and make a pull request against it for things like the semi-major axis. :)

But it would be very satisfying to be able to add as many features to `sgp4` itself, and then simply expose them through Skyfield, instead of doing satellite math up in the Skyfield code instead of letting the satellite code take care of it.

Take a look, and let me know what you think — thanks! :)